### ISSUE TITLE
Error when converting pt file to onnx after packaging the project to exe with pyinstaller

### ISSUE THREAD

**Issue by Harlan2001** [ultralytics/ultralytics#20379]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

When I package my project into an exe using the command pyinstaller -D main.py and run the exe, then click "Convert" on the GUI to convert a pt file to an onnx file, the following error appears:

### Source code:

```
model = YOLO(pt_file)
model.export(format="onnx", imgsz=[640, 640], opset=12)
```

After running the packaged exe, the console shows the following message:

> requirements: Ultralytics requirements ['onnx>=1.12.0', 'onnxslim', 'onnxruntime'] not found, attempting AutoUpdate...

These three libraries are already installed:

```
onnx                      1.16.1
onnxruntime               1.21.1
onnxslim                  0.1.50
ultralytics               8.3.118
```

### Environment

Ultralytics 8.3.118  Python-3.12.10 torch-2.7.0+cpu CPU (12th Gen Intel Core(TM) i7-1255U)
YOLOv8n summary (fused): 72 layers, 3,151,904 parameters, 0 gradients, 8.7 GFLOPs

PyTorch: starting from 'D:\SaejongTrainer\SaejongFramework\MyDetect\yolov8n.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 84, 8400) (6.2 MB)
requirements: Ultralytics requirements ['onnx>=1.12.0', 'onnxslim', 'onnxruntime'] not found, attempting AutoUpdate...

### test.py
```
import torch
import torchvision

model = torchvision.models.resnet18(pretrained=True)
torch.save(model.state_dict(), "resnet18.pt")
model.load_state_dict(torch.load("resnet18.pt"))
model.eval()
torch.onnx.export(model, torch.randn(1, 3, 224, 224), "model.onnx")
```

Then, run

> pyinstaller test.py

Then,test the exe file

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Harlan2001, thank you for bringing this to our attention and for providing detailed information! 🚀 This is an automated response to help you get started while an Ultralytics engineer reviews your issue.

We recommend checking our [Docs](https://docs.ultralytics.com/) for additional guidance, especially the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as many common questions are addressed there.

Since this is a 🐛 Bug Report related to packaging with PyInstaller and ONNX export, please ensure your [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) isolates the issue as much as possible. This will help us debug more efficiently.

You’re welcome to join the Ultralytics community for more discussion and support:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth Q&A: [Discourse](https://community.ultralytics.com/)
- Share knowledge: [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Please verify your issue persists after upgrading to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date verified environments (with dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

An Ultralytics engineer will review your report and follow up soon. Thank you for helping us improve YOLO!

**Comment by Y-T-G** [comment]
This seems like a PyInstaller issue. Not due to Ultralytics.
